src/FederationSetup/GenesisMiner.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using Stratis.Bitcoin.Features.SmartContracts.PoA;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace FederationSetup
+{
+    public class GenesisMiner
+    {
+        private readonly ITestOutputHelper output;
+
+        public GenesisMiner(ITestOutputHelper output = null)
+        {
+            if (output == null) return;
+            this.output = output;
+        }
+
+        //[Fact]
+        [Fact(Skip = "This is not a test, it is meant to be run upon creating a network")]
+        public void Run_MineGenesis()
+        {
+            var consensusFactory = new SmartContractPoAConsensusFactory();
+            string coinbaseText = "https://www.coindesk.com/apple-co-founder-backs-dorsey-bitcoin-become-webs-currency/";
+            this.output.WriteLine(this.MineGenesisBlocks(consensusFactory, coinbaseText));
+        }
+
+        public string MineGenesisBlocks(SmartContractPoAConsensusFactory consensusFactory, string coinbaseText)
+        {
+            var output = new StringBuilder();
+
+            Console.WriteLine("Looking for genesis blocks for the 3 networks, this might take a while.");
+            Console.WriteLine(Environment.NewLine);
+
+            var targets = new Dictionary<uint256, string>
+            {
+                { new uint256("00000fffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"), "-- MainNet network --" },
```

Comment:
@YakupIpek can you check as @StratisIain has changed this values recently.

Reply:
It is done.